Hazelcast.Net/Hazelcast.IO.Serialization/HazelcastJsonValue.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+// http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Hazelcast.Util;
+using System;
+
+namespace Hazelcast.IO.Serialization
+{
+    /// <summary>
+    ///  Class for serializing/deserializing Hazelcast JSON types
+    /// </summary>
+    public class HazelcastJsonValue
+    {
+        private string _jsonString;
+
+        internal HazelcastJsonValue(string json)
```

Comment:
this should be private to force static method usage that does a null validation from everywhere

Reply:
We use this constructor in default serializers class also (so the same lib) -  DefaultSerializers.cs:53

Reply:
Changed to factory method